src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -875,6 +930,16 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
         return { x: defaultPositionX, y: defaultPositionY };
     };
 
+    const onNodeMouseEnter = (evt, node) => {
+        currentHoveredRef.current = node.id;
+        setElements([...elements]);
+    };
+
+    const onNodeMouseLeave = (evt) => {
+        currentHoveredRef.current = '';
+        setElements([...elements]);
```

Comment:
Why do the elements need to be reset in this scenario? They haven't changed?

If we need to trigger a re-render then perhaps we need to make the component on the ref too 🤔

Reply:
I need to make the re-render because the component the mouse event is leaving is indeed changing but that change is not visualized until render is done, what would you recomend

Reply:
If the styling - or layout - is conditional on the value of currentHoveredRef, then it should be a state property rather than a ref. Changes to state will trigger the appropriate update to the layout.